src/dune_rules/pkg_rules.ml

```diff
@@ -1198,16 +1205,37 @@ end = struct
           let doc =
             Path.outside_build_dir @@ Path.Outside_build_dir.relative prefix "doc"
           in
+          let* build_command =
+            match build_command with
+            | None | Some Dune -> Memo.return build_command
+            | Some (Action action) ->
+              let+ installed = Fs_memo.dir_exists prefix in
```

Comment:
Note that doing this check at rule generation time at is wrong as I've mentioned. Doesn't prevent this PR from being merged, but you should work on pushing these checks to action execution.